### ISSUE TITLE
[CR] Port Windows console to use libtcod/SDL instead of GDI.

### ISSUE THREAD

**Issue by karlnp** [CleverRaven/Cataclysm-DDA#12974]
We spend the vast, vast majority of our time drawing to the console or waiting to draw to the console (or spinning while waiting for input?), as seen here:

![Windows profiling stack](http://puu.sh/j2KGG/2501d619e8.png)

It makes sense that we spend a lot of time in the drawing functions, but we drastically underuse our CPU cycles, and there's a lot of room for improvement:

![Windows profiling stack with hotspots](http://puu.sh/j2KVr/cd29916dbb.png)

So, given that using DirectX/hardware acceleration is a "[nice to have for later](http://smf.cataclysmdda.com/index.php?topic=10801.0)" and we already use SDL for the tiles build, I'm porting wincurses.c to wincurses_[libtcod](https://bitbucket.org/libtcod/libtcod).c on [my fork](https://github.com/karlnp/Cataclysm-DDA). Once manual testing is done I'll try and minimize @narc0tiq 's workload on adding a new Jenkins build target.

This will add a couple MB to required libs to build on Windows, but will probably also make building on Windows easier (as long as you can get SDL to work, which isn't always that fun.)
##

**Comment by narc0tiq** [comment]
I think we wanted libtcod as an option at one point, but nobody was willing to go through with it at the time (before the SDL builds, even). It certainly sounds like a better choice for the wincurses build to me.

**Comment by karlnp** [comment]
The Windows curses code might be the oldest code in the codebase - Kevin's reaction to our rendering techniques was "Jesus Christ," and then "That's why it's so slow!"

**Comment by karlnp** [comment]
Here's the profiling info if you want to poke around. https://gist.github.com/karlnp/a2152a32cf313315d282

**Comment by karlnp** [comment]
```
[11:27] <MacrosBlack> I say go for it!
[11:28] <karlnp> i figured you'd be behind the idea :P
[11:29] <MacrosBlack> I hate the 'ain't broke don't fix it' mentality
[11:30] <karlnp> it's so bad currently. like it's astounding. like i'm having trouble porting it because it's so old
[11:31] <MacrosBlack> which is because it's years of "Well, it still mostly works, I won't touch it"
[11:37] <karlnp> and almost everyone devs/plays on linux or mac i think
[11:37] <karlnp> whereas i am a contrary asshole
[11:38] <karlnp> i uploaded profiling info if you wanted to check it out https://gist.github.com/karlnp/a2152a32cf313315d282
[11:38] <MacrosBlack> I think there are more windows players
[11:38] <karlnp> well. of the devs, i mean
[11:38] <MacrosBlack> just fewer devs/contributors
[11:42] <Narc0tiq> wincurse is just a pain; nobody understood it when we were back at v0.3, and I doubt anyone really understands it now.
[11:43] <Narc0tiq> But without Windows devs willing to take it on, it's not going to change.
[11:45] <MacrosBlack> Mostly because the Windows dev environment for cdda is arcane and hostile
[11:45] <karlnp> i understand it now, and i hate it
[11:45] <karlnp> throw it In The Garbage
[11:46] <MacrosBlack> wincurses or the windows dev environment?
[11:46] <Narc0tiq> Filing it in the circular bin isn't a bad idea at all.
```

**Comment by drbig** [comment]
+1 for making it saner on Windows. As much as I stay away, sanity does count.

**Comment by kevingranade** [comment]
As an option, absolutely, as the primary renderer for console on windows,
it's just a matter of compatibility.  If it runs everywhere and is faster,
that's not a hard call to make.  I don't like extra dependencies, but
frankly it's up to the windows devs what dependencies to have on windows.

**Comment by karlnp** [comment]
I don't like them either, and DirectX comes with every version of Windows, but DirectX also depends on hardware acceleration, which we've already ruled out. There's GDI+ maybe but it's not that much newer than GDI itself and is redirected to DirectX anyway under the hood. I'm not sure what to recommend, honestly.

**Comment by kevingranade** [comment]
On Jul 17, 2015 6:29 PM, "Karl Parakenings" <notifications@github.com>
wrote:
>
> I'm not sure what to recommend, honestly.

Ok, I'll take a swing at that softball.
"Linux, ba-dum-tish."

**Comment by karlnp** [comment]
😓

**Comment by illi-kun** [comment]
### This issue was closed as it appears inactive.

Reducing open issues to those which are (or will) be actively worked upon helps us focus our efforts. This issue has not been deleted - it still appears in searches and if it contains relevant information you are encouraged to continue to link to it.

### If this issue was a bug
It should be reopened if it can be reproduced in the current build. You can obtain the most recent copy [here](http://dev.narc.ro/cataclysm/jenkins-latest/). Please check there is not a more recent report of this bug before doing so. If no more recent report exists you should continue the discussion in this issue.

### If this was a feature request
If the consensus was that the idea was good you could consider submitting an implementation via a PR. If you want to comment further please do so here as opposed to opening a new issue. Before posting check nobody has already made the same point and consider whether your comments are likely to lead to an implementation. If you have doubts about either consider instead [voting](http://iv.cataclysmdda.com:8111/all) for the issue

### If you want to work on this issue
Then either assign it to yourself or if you are unable to do so claim it via adding a comment. Please don't assign others or make a general request for action.
